### PR TITLE
Cut: initial implementation

### PR DESCRIPTION
**Pull request by mkindahl**
Implementation of Linux cut(1) with some differences:

* Linux cut(1) does not have a short option name for `--output-delimiter` so I picked `-D` (since short option for `--input-delimiter` is `-d`.

* Error messages are in some cases different.

### CODE REVIEW

**Comment by mkindahl** [comment]
Sorry about the time. I did not have as much free time as I expected.

Some questions on how you want to proceed on these:

**Tests.** I have tests for the `Range` and `RangeSet`, but I am a little unsure how you want to have tests for the rest of the utility.

* I could create unit tests for each of the functions, but that might require some re-writes of the code to make it possible to create in-line examples. Probably the easiest to create, but will not cover all corner-cases.
* Another option is to create a bunch of files as examples and execute shell commands with them. It would be a more thorough test, but will take more time, and I see no support for writing those kinds of tests.

**Structure.** Right now, everything is in a single file, but the `Range` and `RangeSet` data types could be split out into a separate file. I haven't seen other examples doing that, so not sure how you want to have that.

File: cut/src/cut.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: cut
+version: "0.0.0"
+author: Mats Kindahl <mats.kindahl@gmail.com>
+about: "Print selected parts of lines from each FILE to standard output.\n\nWith no FILE, or when FILE is -, read standard input."
```
Comment (GrayJack) [inline cut/src/cut.yml]: Always use the word "display" over "print", just for stylistics and consistency with other tools, there is other cases on line 41 and 53

File: cut/src/cut.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+args:
+    - files:
```
[the remark this answers is not in the document]
Comment (mkindahl) [reply cut/src/cut.yml]: Done.

File: cut/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+
+    /// Macro to assert that an expression matches a pattern.
+    macro_rules! assert_matches {
+        ($e:expr, $p:pat) => {
+            if let $p = $e { true } else { false }
+        };
+    }
+
+    #[test]
+    fn test_range() {
```
[the remark this answers is not in the document]
Comment (mkindahl) [reply cut/src/main.rs]: Done.

File: cut/src/cut.yml (hunk 1 above)
Comment (mkindahl) [reply cut/src/cut.yml]: Done.

**Comment by mkindahl** [comment]
> Changes requested are mostly stylistic changes.
> 
> **About tests**
> 
>     * No need to implement test for every single function, for some cases I'm not even sure how it's possible to create unit tests (like some stuff that I do on `coreutils_core` that is highly dependent on the system configurantion, but maybe, for your case, you can do like `wc` tests (not sure though)

I'll look at the `wc` tests later, but haven't done that now.

> 
>     * Yeah, that really shows the need to create a infrastructure to test in shell the commands
> 
>     * Could you add the tool on other *.toml workspaces, so the CI can test the building on other platforms?

Did that. Saw some errors, and fixed some things, so I will investigate and make sure there are no more.
 
> 
> 
> **About structure**
> 
> I think it's ok with this amount of lines, , but if you think it would be better divided, fell free to create a module(s) for it.

Kept it as it is. I might split it up into modules later, but right not it is quite clear as it is.

> I think it's better to have the `main` function as the first thing after `use` statements, everything else comes after

Moved the `main` to first.

Think it should be complete now. PTAL.
